projects/common/Database/DatabaseConnector.cpp

```diff
@@ -83,7 +83,8 @@ void DatabaseConnector::Borrow(DATABASE_TYPE type, std::function<void(std::share
         _connectorPools[type].add(std::unique_ptr<DatabaseConnector>(newConnector));
     }
 
-    func(std::shared_ptr<DatabaseConnector>(_connectorPools[type].acquire()));
```

Comment:
This was a weird bug, but I get the impression that it points to a wider design problem with this class? At least that is what I was getting from the Discord conversation.

In any case, this appears to patch over it for this pull request.

Reply:
This is just supposed to call the callback function "func" while passing a database connector into it. I am unsure which part of this fails, you should check if calling func with modified parameters to not pass anything into it is allowed, or maybe if doing the .aquire() outside of the parameters is allowed.

Reply:
I'm not quite certain what is meant by this, would you be able to explain this to me at some point?

Reply:
https://github.com/novuscore/NovusCore/blob/master/projects/common/Database/DatabaseConnector.cpp#L86

This is what Pursche meant